libraries/core_libs/storage/include/storage/storage.hpp

```diff
@@ -143,6 +143,7 @@ class DbStorage : public std::enable_shared_from_this<DbStorage> {
   std::atomic<bool> snapshots_enabled_ = true;
   const uint32_t kDbSnapshotsMaxCount = 0;
   std::set<PbftPeriod> snapshots_;
+  std::vector<vote_hash_t> reward_votes_keys_;
```

Comment:
if you put this cache into VoteManager, we would not need to call getRewardVotes() from db, which is still using db iterator but it could use this in memory vector. getRewardVotes() would be called only during node startup to init this cache

Reply:
I have rewritten this PR, please review all of it again